modules/040-node-manager/crds/nodegroupconfiguration.yaml

```diff
@@ -0,0 +1,57 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: nodegroupconfigurations.deckhouse.io
+  labels:
+    heritage: deckhouse
+    module: node-manager
+spec:
+  group: deckhouse.io
+  scope: Cluster
+  names:
+    plural: nodegroupconfigurations
+    singular: nodegroupconfiguration
+    kind: NodeGroupConfiguration
+  preserveUnknownFields: false
+  versions:
+    - name: v1alpha1
+      served: true
+      storage: true
+      schema: &schema
+        openAPIV3Schema:
+          type: object
+          required:
+            - spec
+          properties:
+            spec:
+              type: object
+              required:
+                - content
+              properties:
+                content:
+                  type: string
+                  description: The content of the step, can be either go template or plain bash.
+                weight:
+                  type: integer
+                  default: 100
+                  description: Order of the step execution.
+                nodeGroups:
+                  type: array
+                  description: List of NodeGroups to apply the step for.
+                  default:
+                    - "*"
```

Comment:
All node groups are affected by default? Maybe it would be more explicit to make this field required? If this field were required, users would think about which node groups to apply the custom step.